tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassLoaderRunnerConfig.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.junit4.runners;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Specifies a configuration needed by {@link ArtifactClassloaderTestRunner} in order to
+ * run the test.
+ *
+ * @since 4.0
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+public @interface ArtifactClassLoaderRunnerConfig
+{
+
+    /**
+     * @return a comma separated list of packages to be added as PARENT_ONLY for the
+     * container class loader, default (and required) packages is empty.
+     * In case of having to append one you should also include the default list.
+     */
+    String extraBootPackages() default "";
+
+    /**
+     * @return {@link String[]} with the extension {@link Class} name of the extensions that would be used to create and load it using a plugin/extensio
+     * {@link ClassLoader}. If no extensions are defined the plugin/extension class loaders will not be created.
+     */
+    String[] extensions() default {};
```

Comment:
It is important to document here that the value cannot be a FooExtension.class.getName() reference as it will fail to create the test instance if that class is not exported in the extension.